utils/check-exists.js

```diff
@@ -0,0 +1,23 @@
+const db = require("../db/connection");
+
+const checkArticleExists = (article_id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1;", [article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Article not found" });
+      }
+    });
+};
+
+const checkUserExists = (username) => {
```

Comment:
author in the comments table is a foreign key which relates to username in the user table - so therefore it will violate if the user does not exist! I don't think you need this util function! 

(same with checkArticleExists!)